dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -233,15 +234,20 @@ case class BinaryOutcomeDLCClient(
   def createFundingTransactionSigs(): Future[FundingSignatures] = {
     val fundingTx = createUnsignedFundingTransaction
 
-    val sigFs = fundingUtxos.foldLeft(Vector.empty[Future[PartialSignature]]) {
+    val sigFs = fundingUtxos.foldLeft(
+      Vector.empty[Future[(TransactionOutPoint, PartialSignature)]]) {
       case (vec, utxo) =>
-        val sigF = BitcoinSignerSingle.signSingle(utxo,
-                                                  fundingTx,
-                                                  isDummySignature = false)
+        val sigF = BitcoinSignerSingle
+          .signSingle(utxo, fundingTx, isDummySignature = false)
+          .map(sig => (utxo.outPoint, sig))
         vec :+ sigF
     }
+    val sigsF = Future.sequence(sigFs)
 
-    Future.sequence(sigFs).map(FundingSignatures.apply)
+    val sigMapF = sigsF.map { sigs =>
+      sigs.map(sig => (sig._1, sigs.filter(_._1 == sig._1).map(_._2))).toMap
+    }
+    sigMapF.map(FundingSignatures.apply)
```

Comment:
Rename to `outPointAndSigF` and all other names accordingly, then use `_.map { case (outPoint, sig) => ...` at the bottom, this is pretty hard to read as is